IPython/core/completer.py

```diff
@@ -472,6 +472,18 @@ def _safe_isinstance(obj, module, class_name):
     return (module in sys.modules and
             isinstance(obj, getattr(__import__(module), class_name)))
 
+def _safe_really_hasattr(obj, name):
```

Comment:
Does it make sense to put this phony-method canary in safe_hasattr itself? I seem to recall that we also have this logic in core.formatters.